writer.go

```diff
@@ -1103,8 +1107,9 @@ func (wb *writerBuffers) swapPageAndScratchBuffers() {
 }
 
 type writerColumn struct {
-	pool  BufferPool
-	pages []io.ReadWriteSeeker
+	pool       BufferPool
+	pageBuffer io.ReadWriteSeeker
```

Comment:
Note: one non-obvious change here is that the pages seemed to hold an invariant that the readers/writers would seek back to zero when finished, but this PR changes this to force consumers to seek to zero before reading.